app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -279,8 +279,21 @@ object AppRepository {
         highlightsDatabaseRepository.insert(values, lecture.lectureId)
     }
 
-    fun readLectureByLectureId(lectureId: String) =
-            lecturesDatabaseRepository.queryLectureByLectureId(lectureId).first().toLectureAppModel()
+    fun readLectureByLectureId(lectureId: String): Lecture {
+        val lecture = lecturesDatabaseRepository.queryLectureByLectureId(lectureId).toLectureAppModel()
+
+        val highlight = highlightsDatabaseRepository.queryByEventId(lectureId.toInt())
+        if (highlight != null) {
+            lecture.highlight = highlight.isHighlight
+        }
+
+        val alarms = alarmsDatabaseRepository.query(lectureId)
```

Comment:
Read alarms because `FahrplanFragment.loadLectureList()` is doing it here:
https://github.com/EventFahrplan/EventFahrplan/blob/0cd0820a8c12d832a8b2e57d937d72da2a4110c0/app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java#L771